presto-hive/src/main/java/com/facebook/presto/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -547,6 +550,34 @@ public synchronized void alterPartition(String databaseName, String tableName, P
     }
 
     @Override
+    public synchronized void createRole(String role, String grantor)
```

Comment:
Why not create a synchronized object shared between roles methods instead of having synchronized method?
You don't interfere with e.g. `getPartitionNames` or `alterPartition` so you don't need synchronization with all synchronized methods in this class.